spec/controllers/spree/admin/variants_controller_spec.rb

```diff
@@ -35,6 +35,84 @@ module Admin
           assigns(:variants).should match_array [v1, v2]
         end
       end
+
+      describe '#destroy' do
+        let(:variant) { create(:variant) }
+
+        context 'when requesting with js' do
+          before do
+            allow(Spree::Variant).to receive(:find).with(variant.id.to_s) { variant }
+            allow(variant).to receive(:delete)
+          end
+
+          it 'deletes the variant' do
+            spree_delete :destroy, id: variant.id, product_id: variant.product.permalink, format: 'js'
+            expect(variant).to have_received(:delete)
+          end
+
+          it 'shows a success flash message' do
+            spree_delete :destroy, id: variant.id, product_id: variant.product.permalink, format: 'js'
+            expect(flash[:success]).to eq(I18n.t('notice_messages.variant_deleted'))
+          end
+
+          it 'renders spree/admin/shared/destroy' do
+            spree_delete :destroy, id: variant.id, product_id: variant.product.permalink, format: 'js'
+            expect(response).to render_template('spree/admin/shared/_destroy')
+          end
```

Comment:
why are these 3 specs above not one spec with 3 assertions? is it your testing style decision? I am curious...

Reply:
Yep, stems from http://www.betterspecs.org/#single and the conventions I used to work with in previous teams I guess.

Reply:
ah, ok. but that's for unit testing, right? this is controller level testing, and in this case, it doesnt bring much added readability.
I am just trying it out to find out what I prefer. I think I prefer this faster shorter version:
```
        context 'when requesting with js' do
          before do
            allow(Spree::Variant).to receive(:find).with(variant.id.to_s) { variant }
            allow(variant).to receive(:delete)
          end

          it 'deletes the variant' do
            spree_delete :destroy, id: variant.id, product_id: variant.product.permalink, format: 'js'

            expect(variant).to have_received(:delete)
            expect(flash[:success]).to eq(I18n.t('notice_messages.variant_deleted'))
            expect(response).to render_template('spree/admin/shared/_destroy')
          end
```

Reply:
thanks for the link!

Reply:
I quite like to have one assertion per test case. When some of them fail, you get a better picture of what's broken and what's still working. They are also different features. You want to delete the variant and you want to inform the user about it. In this case I agree that your version is easier to read. -> trade off :man_shrugging: